### PR TITLE
Create intro tutorial part 3

### PR DESCRIPTION
**Pull request by rwegener2**
### Background
The 3rd and FINAL installment of the introduction tutorial.  Focused on Pangeo Forge Cloud, transitioning the local recipe for a PR.

### Notes
* I don't actually reference the sandbox here. My thinking is that we should type out part 3 to be the steps to creating a recipe, irrespective of where the person chooses to do so. There is a separate task in the project board for adding the sandbox link to the tutorials, so I propose to save the sandbox link for that ticket.
* ~this [staged recipes issue](https://github.com/pangeo-forge/staged-recipes/issues/104) was created to address where an example recipe.py/meta.yml should live~ 

### questions
* ~@cisaacstern @rabernat Is there specific information that someone should include in their PR to staged recipes?~

### CODE REVIEW

File: docs/introduction_tutorial/intro_tutorial_part3.ipynb [hunk 1]
```diff
@@ -0,0 +1,165 @@
+{
```
Comment (rabernat) [inline docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: <p>Recipe is not actually defined here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pangeo-forge/pangeo-forge-recipes/pull/307/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (rabernat) [inline docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: <p>I think we need a lot more detail about what to put in the meta.yaml file. This document should be a useful reference: https://github.com/pangeo-forge/roadmap/blob/master/doc/adr/0002-use-meta-yaml-to-track-feedstock-metadata.md</p><p> </p><p>We should show a completed meta.yaml for this recipe.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pangeo-forge/pangeo-forge-recipes/pull/307/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by rabernat** [comment]
This feels a little thin. I was expecting a more detailed walk through each section of meta.yaml. Let me know what is unclear about meta.yaml and I will do my best to help.

**Comment by rwegener2** [comment]
I'm glad you pushed for more detail @rabernat .  I beefed out many sections, the `meta.yml` section in particular, and I think it's a lot more informative now.

The downside of the expansion is there are several open questions written in.  You can read them in the docs, or all of them together here:
- does the name of the data folder staged recipes hold any significance?
- [Note] I don't think we want to describe every possible detail or configuration of the meta.yml in the intro tutorial. It would eventually be nice to have a page in the Cloud Automation User Guide where this is done in detail which can be referenced in the tutorial.
- what it the `recipes:` section?
- how are `roles` and `license` chosen under `providers`?
- what are the required fields for a `maintainer`?
- how does someone know what bakeries are available to them? Can anyone just request to host their data on any available bakery?

Cool thing to note: I went with an idea @cisaacstern had a week or so ago and followed the style of [these docs](https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/) in adding a full file preview as we walk through the `meta.yml`. Excited to hear what you think!

**Comment by rabernat** [comment]
> * does the name of the data folder staged recipes hold any significance?

The name of the folder will be turned into the name of the feestock. E.g. if the folder name is `noaa-oisst`, when the PR is merged, a repo called `noaa-oisst-feedstock` will be created.

> * I don't think we want to describe every possible detail or configuration of the meta.yml in the intro tutorial. It would eventually be nice to have a page in the Cloud Automation User Guide where this is done in detail which can be referenced in the tutorial.

👍 💯 

> * what is the `recipes:` section?

From https://github.com/pangeo-forge/roadmap/blob/master/doc/adr/0002-use-meta-yaml-to-track-feedstock-metadata.md

"The recipes section explains what recipes are contained in the repo. The recipes are defined in python code in a .py file (python module)."

> * how are `roles` and `license` chosen under `providers`?

These concepts are copied verbatim from https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#provider-object.

We need to emphasize the license aspect. It is up to the user to understand the license of the data they want to ingest. And we need to expose this license clearly on the website.

> * what are the required fields for a `maintainer`?

name and github should be required. email and orcid should be optional.

> * how does someone know what bakeries are available to them? Can anyone just request to host their data on any available bakery?

They go to https://pangeo-forge.org/bakeries (link not working yet). For now there is only one bakery, so we can just provide a default.

**Comment by rwegener2** [comment]
Thanks for all the clarifications @rabernat!  I updated and re-requested review.

Two notes are left commented in-line:
- Is `columbia-staging` the bakery ID to be using in the tutorial?
- description of `object: "recipe:recipe"` was vague, I'm interested in your feedback there

File: docs/introduction_tutorial/intro_tutorial_part3.ipynb [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
```
Comment (rabernat) [inline docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: <p>The first <code>recipe</code> refers to the name of the module (<code>recipe.py</code> ). The second <code>recipe</code> refers to the name of the Recipe object inside that module. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pangeo-forge/pangeo-forge-recipes/pull/307/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (rabernat) [inline docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: <p>Charles, will we have a "production" bakery up and running before Friday? What naming conventions should we use for the bakeries?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pangeo-forge/pangeo-forge-recipes/pull/307/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (cisaacstern) [reply docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: <p>I had not been planning on that, but it is theoretically not a huge lift. Should we prioritize that deployment?</p>

Comment (rabernat) [reply docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: If we really want to be doing things right, we will have two versions of everything. "staging" is not the real thing. However, that may not be feasible here.

Comment (rwegener2) [reply docs/introduction_tutorial/intro_tutorial_part3.ipynb:1]: name: "pangeo-ldeo-nsf-earthcube"

**Comment by cisaacstern** [comment]
Noting that I've added `"pangeo-ldeo-nsf-earthcube"` as a bakery to our production database

https://api.pangeo-forge.org/bakeries/1

so contributors will now be able to specify this name in their PRs without confusing @pangeo-forge-bot

**Pull request merged**